apps/protocol/src/app/hooks/useScaledInput.ts

```diff
@@ -0,0 +1,37 @@
+import { createMemo } from 'react-use'
+import { BigDecimal } from '@apps/bigdecimal'
+import { BigDecimalInputValues } from '@apps/hooks'
+import { InputRatios, ScaledInput } from '@apps/types'
+
+export const useScaledInput = createMemo((inputValues: BigDecimalInputValues, inputRatios: InputRatios): ScaledInput => {
+  const touched = Object.values(inputValues).filter(value => value.touched && value.amount.exact.gt(0))
+
+  const highAmounts = touched.map(value => value.amount)
+  const highTotal = BigDecimal.sum(...highAmounts)
+
+  const scaledHighAmounts = touched.map(({ amount, address }) =>
+    // Only scale with a ratio; massets are not scaled
+    inputRatios[address] ? amount.mulRatioTruncate(inputRatios[address]).setDecimals(18) : amount,
+  )
+  const scaledHighTotal = BigDecimal.sum(...scaledHighAmounts)
+
+  return touched.reduce<ScaledInput>(
+    (prev, { amount: high, address }, idx) => {
+      const scaledHigh = scaledHighAmounts[idx]
+
+      const low = high.divPrecisely(highTotal)
+      const lowTotal = prev.lowTotal.add(low)
```

Comment:
PS @rafaelugolini if it's not clear, the low amounts are used to calculate the price impact; high == the amount the user has selected, low == a small amount of the same proportion. If the distance is quite different, that's price impact.

Reply:
The tests should make this clear, I hope 🤞